native_client/deepspeech.cc

```diff
@@ -128,13 +128,23 @@ Model::infer(float* aMfcc, int aNFrames, int aFrameLen)
   Tensor n_frames(DT_INT32, TensorShape({1}));
   n_frames.scalar<int>()() = aNFrames;
 
-  // "Reshape_3" = logits, before the built-in CTC decoder is called.
-  const char* output_node_name = mPriv->scorer == NULL ? "output_node" : "Reshape_3";
+  // The new decoder takes the logits as input, the old one is built into the graph
+  const char* output_node_name = mPriv->scorer == NULL ? "output_node" : "logits";
 
   std::vector<Tensor> outputs;
   Status status = mPriv->session->Run(
     {{ "input_node", input }, { "input_lengths", n_frames }},
     {output_node_name}, {}, &outputs);
+
+  // If "logits" doesn't exist, this is an older graph. Try to recover.
```

Comment:
I tend to remember there is a way to "query" tensors from the graph, can't we rely on that instead of try/error based on Session::Run() ? That would feel nicer IMHO :)

Reply:
I don't know the TensorFlow C++ API very well, but I agree that'd be cleaner, I'll check.

Reply:
The feature I had in mind is Tensorflow's Python `get_tensor_by_name()`, which seems not to have a C++ side :(

Reply:
Yeah, according to this answer from a TensorFlow dev It's Just Strings All The Way Down: https://stackoverflow.com/questions/39378657/c-equivalent-of-python-tf-graph-get-tensor-by-name-in-tensorflow